Evaluant.Calculator/Domain/EvaluationVisitor.cs

```diff
@@ -29,7 +29,26 @@ public override void Visit(LogicalExpression expression)
         private static readonly Type[] CommonTypes = { typeof(Int64), typeof(Double), typeof(Boolean), typeof(String), typeof(Decimal) };
 
         /// <summary>
-        /// Gets the the most precise type.
+        /// Gets the the most precise type of both objects, even if one is null.
+        /// </summary>
+        /// <param name="a">Object a.</param>
+        /// <param name="b">Object b.</param>
+        /// <returns></returns>
+        private static Type GetMostPreciseType(object a, object b)
```

Comment:
I'ts itching me to have .GetType inside a loop. It's not a big deal, it's just my OCD
Does this work?

```        private static Type GetMostPreciseType(Type a, Type b)
        {
            foreach (Type t in CommonTypes)
            {
                if (a == t || b == t)
                {
                    return t;
                }
            }

            return a ?? b;
        }

        public int CompareUsingMostPreciseType(object a, object b)
        {
            Type mpt = options.AllowNullParameter() 
                     // Allow nulls to be compared with other values
                     ? GetMostPreciseType(a?.GetType(), b?.GetType()) ?? typeof(object)
                     // No breaking changes fallback
                     : GetMostPreciseType(a.GetType(), b.GetType());
            return Comparer.Default.Compare(Convert.ChangeType(a, mpt), Convert.ChangeType(b, mpt));
        }
```

Reply:
Yep, that works fine. PR updated.

Reply:
can the overload GetMostPreciseType(object a, object b) be deleted?